ArduCopter/mode_turtle.cpp

```diff
@@ -0,0 +1,138 @@
+#include "Copter.h"
+
+#if MODE_TURTLE_ENABLED == ENABLED
+
+#define CRASH_FLIP_EXPO 35.0f
+#define CRASH_FLIP_STICK_MINF 0.15f
+#define power3(x) ((x)*(x)*(x))
+
+// throw_init - initialise turtle mode controller
+bool ModeTurtle::init(bool ignore_checks)
+{
+    // do not enter the mode when already armed or when flying
+    if (motors->armed() || SRV_Channels::get_dshot_esc_type() == 0) {
+        return false;
+    }
+    // reverse the motors
+    hal.rcout->disable_channel_mask_updates();
+    change_motor_direction(true);
+
+    // arm
+    motors->armed(true);
```

Comment:
Do we really want to arm the motors just by changing into the flight mode?  I think we should force users to arm again if disarmed.

Reply:
Arming checks added